### PR TITLE
benchadapt: asv adapter 

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
Asv (airspeed velocity) benchmarks are used by pandas and many other libraries. 

Ping to [jorisvandenbossche](https://github.com/pandas-dev/pandas/commits?author=jorisvandenbossche) as we talked about this.

### CODE REVIEW

**Comment by DeaMariaLeon** [comment]
When I run "make test" locally I don't see the error that is shown here (test conbenchlegacy). I wonder if I have to debug that or not. If I have to fix that, ideas  would be helpful. Thanks.

**Comment by austin3dickey** [comment]
Thanks for the contribution! I'll review a little bit later 🙂

File: benchadapt/python/benchadapt/adapters/asvbench.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import itertools
+import json
+import os
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List
+
+import numpy as np
```
Comment (austin3dickey) [inline benchadapt/python/benchadapt/adapters/asvbench.py:8]: Would you mind adding `numpy` to `requirements.txt`?

File: benchadapt/python/benchadapt/adapters/asvbench.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+
+class AsvBenchmarkAdapter(BenchmarkAdapter):
+
+    def __init__(
+        self,
+        command: List[str],
+        result_file: Path,
+        benchmarks_file_path: Path,
+        result_fields_override: Dict[str, Any] = None,
```
Comment (austin3dickey) [inline benchadapt/python/benchadapt/adapters/asvbench.py:22]: Could you add the `result_fields_append` arg here too?

File: benchadapt/python/tests/adapters/test_asvbench.py [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+            [["'int64'"], ["None", "'foo'"]],
+            "9837bd38099b9a8813a9b22a59b46e9e9afd7614361947eb24b302192b97eb20",
+        ],
+    },
+    "durations": {},
+    "version": 2,
+}
+
+
+class TestAsvAdapter:
```
Comment (austin3dickey) [inline benchadapt/python/tests/adapters/test_asvbench.py:327]: Thanks for adding these tests!

**Review by austin3dickey — COMMENTED** [review]
Thanks! Looks like the `conbenchlegacy` failure is unrelated to this PR: #1593

File: benchadapt/python/benchadapt/adapters/asvbench.py (hunk 1 above)
Comment (DeaMariaLeon) [reply benchadapt/python/benchadapt/adapters/asvbench.py:8]: Thanks for the review!

File: benchadapt/python/benchadapt/adapters/asvbench.py (hunk 2 above)
Comment (DeaMariaLeon) [reply benchadapt/python/benchadapt/adapters/asvbench.py:22]: Done.

**Comment by DeaMariaLeon** [comment]
@austin3dickey I don't know if I am supposed to ping you when the CI tests are over. I'm not rushing you at all.

**Review by austin3dickey — APPROVED** [review]
LGTM, thanks again!

**Comment by austin3dickey** [comment]
Did you want me to do a release so you can use this from pip?

**Comment by DeaMariaLeon** [comment]
It will be great when that happens (when this can be installed from pip). But it's really up to you. Please do the release when it suits you.

**Comment by DeaMariaLeon** [comment]
And thanks for the review and the merge. 🙂

**Comment by austin3dickey** [comment]
Awesome. `benchadapt==2024.3.20` should be ready to go!

**Comment by DeaMariaLeon** [comment]
Excellent.. thank you!

**Pull request merged**